### DISCUSSION TITLE
Searching for a rule to detect magic string literals

### DISCUSSION THREAD

**Discussion by robopzet** [eslint/eslint#15445] [Q&A] [answered]
For our code I would like to have a check that detects magic string literals, similar to the [no-magic-numbers](https://eslint.org/docs/rules/no-magic-numbers) rule. It should detect all strings not defined in named constants.

Does such a rule exist?

**Comment by aladdin-add** [comment, marked as answer]
no, you'll need to write your own.

**Reply by robopzet** [reply]
OK, thanks.
